client/src/components/Profile/Profile.jsx

```diff
@@ -6,29 +6,52 @@ import { faUserCircle } from "@fortawesome/free-solid-svg-icons";
 import FutureBookings from "../BookingsAll/BookingsAll";
 import { Auth } from "aws-amplify";
 import "./Profile.css";
+import axios from "axios"; 
 
 function Profile() {
   const { user } = useContext(UserContext);
   const [givenName, setGivenName] = useState("");
   const [familyName, setFamilyName] = useState("");
   const [email, setEmail] = useState("");
-  const [phone, setPhone] = useState();
+  const [phone, setPhone] = useState("");
   const [displayInputs, setDisplayInputs] = useState(false);
 
-  useEffect(() => {
-    if (user) {
-      setGivenName(user.attributes.given_name);
-      setFamilyName(user.attributes.family_name);
-      setEmail(user.attributes.email);
-      setPhone(user.attributes.phone_number);
-    }
-  }, [user]);
 
-  // this function should pull from component state and
-  // post to db + update user pool
-  async function updateProfile() {
-   
-  }
+    useEffect(() => {
+        async function fetchUser(){
+          if(user){
+            let req = axios.get(`http://localhost:4000/api/users/${user.attributes.email}`)      
+            let res = await req; 
+            let data = res.data;  
+            setGivenName(data[0].first_name);
+            setFamilyName(data[0].last_name); 
+            setEmail(data[0].email);
+            setPhone(data[0].phone); 
+          }
+      }
+      fetchUser(); 
+  },[user]);
+
+    // this function should pull from component state and
+    // post to db + update user pool
+    async function updateProfile() {
+      axios.patch(`http://localhost:4000/api/users/`, 
+      {
+        "first_name":givenName, 
+        "last_name":familyName,
+        "email":email,
+        "phone":phone
+      }); 
+
+
+      //profile page is updated exept email. 
+      //email shouldn't be changed 
+      user.attributes.given_name=givenName; 
+      user.attributes.family_name=familyName;
+      user.attributes.phone_number=phone; 
```

Comment:
`user` cannot be changed this way. Please remove. We will need to fix this bug later!

Reply:
Thx!